napari_plugin_manager/base_qt_plugin_dialog.py

```diff
@@ -615,6 +627,26 @@ def _action_requested(self):
             if self.action_button.objectName() == 'install_button'
             else InstallerActions.UNINSTALL
         )
+        if (
+            tool == InstallerTools.PIP
+            and action == InstallerActions.INSTALL
+            and self._on_bundle()
```

Comment:
Some folks might not want this behavior, on bundle or not. Let's turn this third condition into `self._warn_pypi()` and then users can choose what to do about it in their subclasses. In our case, `_warn_pypi` will call `on_bundle`.